opencog/moses/scoring/discriminating_bscore.cc

```diff
@@ -195,6 +246,43 @@ vector<discriminator::d_counts> discriminator::counts(const combo_tree& tr) cons
     return ctr_seq;
 }
 
+vector<discriminator::d_counts> discriminator::counts(const Handle& program) const
+{
+    std::vector<d_counts> ctr_seq;
+
+    atomese::Interpreter interpreter(moses::compressed_value_key);
+    ValuePtr _result = interpreter(program);
+    auto link_result = LinkValueCast(_result)->value();
+    int i = 0;
+
+    for (const CTable::value_type& vct : _ctable) {
+        // vct.first = input vector
+        // vct.second = counter of outputs
+
+        double pos = sum_true(vct.second);
+        double neg = sum_false(vct.second);
+        unsigned totalc = vct.second.total_count();
+
+        d_counts ctr;
+        if (HandleCast(link_result.at(i))->get_type() == TRUE_LINK)
```

Comment:
Better use `bool_value_to_bool` define in

https://github.com/singnet/asmoses/blob/998557895ad4e39ea43a80fbef2f5eb75351640c/opencog/utils/valueUtils.h#L12-L15

that way if we change the representation of Boolean values that code won't change.